bundles/org.eclipse.e4.ui.workbench.renderers.swt/src/org/eclipse/e4/ui/workbench/renderers/swt/StackRenderer.java

```diff
@@ -209,6 +217,42 @@ private void addItemToSet(List<CTabItem> itemsToSet, MPart parentParent) {
 		}
 	}
 
+	@Inject
+	@Optional
+	void subscribePerspectiveSwitched(
+			@UIEventTopic(UIEvents.UILifeCycle.PERSPECTIVE_SWITCHED) org.osgi.service.event.Event event) {
+		Object element = event.getProperty(EventTags.ELEMENT);
+		if (element instanceof MPerspective) {
+			setOnboarding((MPerspective) element);
+		}
+	}
+
+	private void setOnboarding(MPerspective perspective) {
+		if (onBoarding == null) {
+			return;
+		}
+
+		onBoarding.getDisplay().asyncExec(() -> {
+			Image image = getImageFromURI(perspective.getIconURI());
+			Image greyImage = new Image(welcomeText.getDisplay(), image, SWT.IMAGE_GRAY);
+			if (welcomeImage.getImage() != null) {
+				welcomeImage.getImage().dispose();
+			}
+			welcomeImage.setImage(greyImage);
+
+			onBoarding.addDisposeListener(e -> {
+				image.dispose();
+				greyImage.dispose();
+				welcomeImage.dispose();
+			});
+
+			String text = NLS.bind("Welcome to {0}", perspective.getLabel()); //$NON-NLS-1$
```

Comment:
as the commit msg says, it is planned to e.g. outsource text, image and maybe actions to an extension point.
Can this text stay NON-NLS until then?

Otherwise this would be the first externalized string in this plugin and would lead to Message.java, messages file and new package for both.